### PR TITLE
docs: add explanation about shebang

### PR DESCRIPTION
**Pull request by hverlin**
See #3491

### CODE REVIEW

File: docs/tasks/toml-tasks.md [hunk 1]
```diff
@@ -282,12 +298,28 @@ ruby = 'latest'
...
 """
 ```
 
 :::
 
+::: details What's a shebang? What's the difference between `#!/usr/bin/env` and `#!/usr/bin/env -S`
+
+A shebang is the character sequence `#!` at the beginning of a script file that tells the system which program should be used to interpret/execute the script.
+The [env command](https://manpages.ubuntu.com/manpages/jammy/man1/env.1.html) comes from GNU Coreutils.
+
```
Comment (hverlin) [inline docs/tasks/toml-tasks.md:306]: Should I indicate here that `mise` is going to create a file and run it instead?
https://github.com/jdx/mise/blob/224a0b523ba63f54b5eaf89a1fc5722188ce3f99/src/cli/run.rs#L375-L383

File: docs/tasks/toml-tasks.md [hunk 2]
```diff
@@ -137,7 +137,13 @@ You can hide a task from the list of available tasks by setting `hide = true`:
...
 [tasks.cleancache]
 run = "rm -rf .cache"
-hide = true # hide this task from the list
+hide = true # hide this task from mise tasks / mise run
+```
+
+Note that if you want to see hidden tasks, you can use the `--hidden` flag:
+
+```shell
+mise tasks ls --hidden
```
Comment (jdx) [inline docs/tasks/toml-tasks.md]: I think we can take this out, it's already documented in the reference and I think easy to figure out

File: docs/tasks/toml-tasks.md [hunk 3]
```diff
@@ -180,20 +186,29 @@ outputs = ['target/debug/mycli']
...
+You can specify a `shell` command to run the script with (default is [`sh -c`](/configuration/settings.html#unix_default_inline_shell_args) or [`cmd /c`](/configuration/settings.html#windows_default_inline_shell_args)):
 
 ```toml
 [tasks.lint]
 shell = 'bash -c'
 run = "cargo clippy"
 ```
 
-or use a shebang:
+or use a `shebang`:
```
Comment (jdx) [inline docs/tasks/toml-tasks.md]: Shebang shouldn't be in backticks

File: docs/tasks/toml-tasks.md (hunk 1 above)
Comment (jdx) [reply docs/tasks/toml-tasks.md:306]: So far I don't believe anyone has needed to be aware of this behavior for any reason so I think no

**Pull request merged**